ha/core/event_analyzer/parser/parser.py

```diff
@@ -159,15 +159,102 @@ class ClusterResourceParser(Parser):
     Subscriber for event analyzer to pass msg.
     """
 
-    def __init__(self):
+    def __init__(self, conf_store=None):
         """
         Init method.
         """
         super(ClusterResourceParser, self).__init__()
         ConfigManager.init("event_analyzer")
+        self._conf_store = conf_store
+        self.cluster_id = Conf.get(const.HA_GLOBAL_INDEX, f"COMMON_CONFIG{_DELIM}cluster_id")
+        self.site_id = Conf.get(const.HA_GLOBAL_INDEX, f"COMMON_CONFIG{_DELIM}site_id")
+        self.rack_id = Conf.get(const.HA_GLOBAL_INDEX, f"COMMON_CONFIG{_DELIM}rack_id")
+        self.predefined_config = {"cluster_id": self.cluster_id, "site_id": self.site_id, "rack_id": self.rack_id}
+        self.system_health_node_key_template = "cortx/ha/system/cluster/$cluster_id/site/$site_id/rack/$rack_id/node/$node_id/health"
         Log.info("ClusterResource Parser is initialized ...")
 
-    def parse_event(self, msg: str) -> HealthEvent:
+    def prepare_key(self, res_alert: dict) -> str:
+        """Prepares a key to be searched in confstore"""
+        replacable_patterns = re.findall(r"\$\w+", self.system_health_node_key_template)
+        for pattern in replacable_patterns:
+            replacable_string = pattern[1:]
+            if replacable_string in self.predefined_config:
+                config_val = self.predefined_config[replacable_string]
+                key = self.system_health_node_key_template.replace(pattern, config_val)
+            else:
+                config_val = res_alert["_resource_name"]
+                key = self.system_health_node_key_template.replace(pattern, config_val)
```

Comment:
Should usage of node_key_template be under check of resource_type == node? This is not strictly related to changes for this PR, but not having resource_type checks will make this node_event_parser not generic resource parser. This can be added later as well, just wanted to check if this is already partially supported using resource_type or not, i.e. doing node event related processing under resource_type == node.

Reply:
I have moved this code to system health. So, no need to use this node_key_template now.